### DISCUSSION TITLE
Can Hermes be used with existing apps that integrate with React Native version 69?

### DISCUSSION THREAD

**Discussion by YOEL311** [reactwg/react-native-new-architecture#112] [Q&A] [answered]
Hey 
Working on existing apps to integrate with React Native

Based on your version 0.69 document, I read it
Hermes engine is not marked as true in your post

[Link ](https://reactnative.dev/docs/0.69/integration-with-existing-apps)

Unlike the 0.71 version
Does it have a purpose?
For existing native applications, is Hermes not recommended in version 69?

[Link](https://reactnative.dev/docs/integration-with-existing-apps)

**Comment by YOEL311** [comment]
Another thing to note

if you mark ``enableHermes:  true`` in ``build.gradle``

that does not work with this code

````
  mReactInstanceManager = ReactInstanceManager.builder()
                .setApplication(getApplication())
                .setCurrentActivity(this)
                .setBundleAssetName("index.android.bundle")
                .setJSMainModulePath("index")
                .addPackages(packages)
                .setUseDeveloperSupport(BuildConfig.DEBUG)
                .setInitialLifecycleState(LifecycleState.RESUMED)
                .build();
````
because ``builder`` constructor call
````
    public static ReactInstanceManagerBuilder builder() {
        return new ReactInstanceManagerBuilder();
    }
````
that call to this constructor  
````
   ReactInstanceManagerBuilder() {
        this.jsInterpreter = JSInterpreter.OLD_LOGIC;
    }
````
after that, the code does not care about ``enebleHermes`` 
````
private JavaScriptExecutorFactory getDefaultJSExecutorFactory(String appName, String deviceName, Context applicationContext) {
        if (this.jsInterpreter == JSInterpreter.OLD_LOGIC) {
            try {
                ReactInstanceManager.initializeSoLoaderIfNecessary(applicationContext);
                JSCExecutor.loadLibrary();
                return new JSCExecutorFactory(appName, deviceName);
            } catch (UnsatisfiedLinkError var5) {
                if (var5.getMessage().contains("__cxa_bad_typeid")) {
                    throw var5;
                } else {
                    HermesExecutor.loadLibrary();
                    return new HermesExecutorFactory();
                }
            }
        } else if (this.jsInterpreter == JSInterpreter.HERMES) {
            HermesExecutor.loadLibrary();
            return new HermesExecutorFactory();
        } else {
            JSCExecutor.loadLibrary();
            return new JSCExecutorFactory(appName, deviceName);
        }
    }
````
if you want to use Hermes you must call  ``setJsEngineAsHermes``
````
mReactInstanceManager = ReactInstanceManager.builder()
                 ...
                .setJsEngineAsHermes(true)
                .build();
````

**Comment by cortinico** [comment, marked as answer]
This working group is reserved to discussions which are related to the React Native New Architecture.

I'm locking this discussion as it's not related to New Architecture and we're trying to keep this space focused only on New Architecture.

To shortly answer your question: the reason why hermes is defaulted to `false` in 0.69, and do true in 0.70+ is because we enabled Hermes by default in RN 0.70. More on this in the announcement blogpost:
https://reactnative.dev/blog/2022/07/08/hermes-as-the-default
